text/0000-localizing-api-messages.md

```diff
@@ -0,0 +1,164 @@
+* RFC PR: 
+  * Preliminary Review: https://github.com/folio-org/rfcs/pull/3
+  * Draft Review : https://github.com/folio-org/rfcs/pull/3
+  * Public Review :
+  * Final Review :
+* FOLIO Issue:
+* Current Status: :no_entry_sign: DRAFT REVIEW :no_entry_sign:
+* Submitter : Radhakrishnan Gopalakrishnan (rgopalakrishnan@ebsco.com)
+* Co-Submitter(s) : Zak Burke (zburke@ebsco.com)
+* Sub Group : 
+  * Marc Johnson (marc.johnson@k-int.com)
+  * Julian Ladisch (julian.ladisch@gbv.de)
+  * Peter Murray (peter@indexdata.com)
+  * Tod Olson (tod@uchicago.edu)
+
+# Localizing API (Backend) Messages
+
+## Summary
+  The purpose of this RFC is to define an approach that allows API developer to provide
+messages in the API response depending on the language preference set in the request. 
+
+
+## Motivation
+
+- Help users understand the information coming from the application backend using a language they understand better
+- To allow FOLIO platform to be used by users across the world 
+- To create a tremendous opportunity for growth that we can never could have achieved within just one country.
+- To Promote consistency and make it easier for tools and translators when handling translation files
+
+### Terminology
+* Base Language - Language on which all other translations are based.
+* Message Key - Refers to the string that is used in place of a hardcoded message in the code
+* Message Value - Refers to the string that is fetched using a message key lookup for a given locale
+* Translation file - Contains Message key and values for a given locale
+* Static Messages - Refers to the messages values stored in the translation file
+* Dynamic Messages - Messages stored in the database that requires translation. A.k.a Runtime messages
+* Message Format - Refers to the standard used for formatting the message values in the translation file
+
+### In Scope Requirements/Use cases
+- Return localized messages based on the value passed in the accept-language header
+- Handle static messages with placeholder(s)
+
+### Out of Scope Requirements/Use cases
+- Naming convention to be used for keys
+- Process for managing (decoupled from lokalise.com) translations
+- Returning formatted (HTML/Markdown) messages
+- Usage of [soft hyphen](https://wiki.folio.org/display/I18N/How+To+translate+FOLIO#HowTotranslateFOLIO-Softhyphentobreakwords) to break messages
+- Support customization per tenant
+- Process  for [back porting](https://wiki.folio.org/display/I18N/Backport) API/Backend messages similar to what we have for the front end 
+- Support for [controlled vocabulary](https://issues.folio.org/browse/UXPROD-3148) (runtime data/data coming from DB tables. E.g. Patron Groups for tenants)
+
+## Detailed Explanation/Design
+#### API Protocol
+* accept-language header value MUST be in ll-CC format. , where ll is a two-letter language code,
+  and CC is a two-letter country code. Refer to [IETF Language Tag standard](https://en.wikipedia.org/wiki/IETF_language_tag)
```

Comment:
Why do we forbid tree letter IETF language codes like dsb, gsw, hsb, kok, moh, nso, prs, quc, quz, sah, sma, smj, smn, sms, syr, tzm?